### PR TITLE
Allow associated storage when using a staked factory

### PR DESCRIPTION
**Pull request by hazim-j**
This PR adds two new test cases to reflect recent changes which allow entities to access associated storage if the userOp creates a new account using a staked factory.

From the latest EIP changes:
> If the UserOp doesn't create a new account (that is initCode is empty), or the UserOp creates a new account using a
  staked `factory` contract, then the entity may also use storage associated with the sender

### CODE REVIEW

**Comment by derekchiang** [comment]
@forshtat @drortirosh could you review this?

File: tests/bundle/test_storage_rules.py [hunk 1]
```diff
@@ -274,6 +289,13 @@ def build_userop_for_factory(w3, entrypoint_contract, factory_contract, rule):
         with_initcode(build_userop_for_sender),
         assert_error,
     ),
+    StorageTestCase(
```
Comment (drortirosh) [inline tests/bundle/test_storage_rules.py:292]: this test is correct with the latest EIP

File: tests/bundle/test_storage_rules.py [hunk 2]
```diff
@@ -132,6 +140,13 @@ def build_userop_for_factory(w3, entrypoint_contract, factory_contract, rule):
         with_initcode(build_userop_for_paymaster),
         assert_error,
     ),
+    StorageTestCase(
```
[the remark this answers is not in the document]
Comment (hazim-j) [reply tests/bundle/test_storage_rules.py:143]: Thanks for the clarification @drortirosh. I'll push a change to invert this test then.

[the remark this answers is not in the document]
Comment (hazim-j) [reply tests/bundle/test_storage_rules.py:143]: fixed

[the remark this answers is not in the document]
Comment (drortirosh) [reply tests/bundle/test_storage_rules.py:143]: sorry. the real problem is not yours, but ours... 
the StorageTestCase is currently not built for adding a rule to the ACCOUNT, and staking a FACTORY...
All tests are built that a single "entity under test" has rules and stakes. I'm now modifying the test to fix for that.

**Pull request closed without merging**